python/tests/unit/test_hostglob_matches.py

```diff
@@ -0,0 +1,59 @@
+import logging
+import sys
+
+import pytest
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s test %(levelname)s: %(message)s",
+    datefmt='%Y-%m-%d %H:%M:%S'
+)
+
+logger = logging.getLogger("ambassador")
+
+from ambassador.ir.irutils import hostglob_matches
+
+@pytest.mark.compilertest
+def test_hostglob_matches():
+    for v1, v2, wanted_result in [
+        ( "a.example.com",     "a.example.com",    True  ),
+        ( "a.example.com",     "b.example.com",    False ),
+        ( "*",                 "foo.example.com",  True  ),
+        ( "*.example.com",     "a.example.com",    True  ),
+        ( "*example.com",      "b.example.com",    True  ),
+        # This is never OK: the "*" can't match a bare ".".
+        ( "*example.com",      ".example.com",     False ),
+        # This is OK, because DNS allows names to end with a "."
+        ( "foo.example*",      "foo.example.com.", True  ),
+        # This is never OK: the "*" cannot match an empty string.
+        ( "*example.com",      "example.com",      False ),
+        ( "*ple.com",          "b.example.com",    True  ),
+        ( "*.example.com",     "a.example.org",    False ),
+        ( "*example.com",      "a.example.org",    False ),
+        ( "*ple.com",          "a.example.org",    False ),
+        ( "a.example.*",       "a.example.com",    True  ),
+        ( "a.example*",        "a.example.com",    True  ),
+        ( "a.exa*",            "a.example.com",    True  ),
+        ( "a.example.*",       "a.example.org",    True  ),
+        ( "a.example.*",       "b.example.com",    False ),
+        ( "a.example*",        "b.example.com",    False ),
+        ( "a.exa*",            "b.example.com",    False ),
+        # '*' has to appear at the beginning or the end, not in the middle.
+        ( "a.*.com",           "a.example.com",    False ),
+        # TLS SAN and RFC4592 disagree here about whether '*' can cross subdomain boundaries.
+        # hostglob_matches() follows RFC4592, since that's what Envoy seems to do when routing.
```

Comment:
I don't think we follow RFC 4592. I think this should say "Regardless of what any RFCs say, hostglob_matches() does what Envoy does, and Envoy allows wildcards to span multiple labels."